frontends/concrete-python/examples/pir/PIR.ipynb

```diff
@@ -0,0 +1,576 @@
+    "def get_ith_element_of_database(one_hot_vector: np.ndarray, database: np.ndarray) -> int:\n",
+    "    return np.dot(one_hot_vector, database)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "c67607e7-a93f-4755-8226-17035c2667bf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "database_length = 8\n",
+    "database = np.array([3, 5, 21, 7, 11, 13, 2, 17])\n",
+    "database_output_bits = np.ceil(np.log2(np.max(database))).astype(np.int32)\n",
+    "\n",
+    "assert database_output_bits == 5\n",
+    "\n",
+    "assert get_ith_element_of_database(make_one_hot_vector(0, size=database_length), database) == database[0]\n",
+    "assert get_ith_element_of_database(make_one_hot_vector(3, size=database_length), database) == database[3]\n",
+    "assert get_ith_element_of_database(make_one_hot_vector(4, size=database_length), database) == database[4]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6ef38c1a-eab5-419a-bea0-aaaf1c66a6aa",
+   "metadata": {},
+   "source": [
+    "## Private information retrieval with FHE\n",
+    "\n",
+    "Now, let's make that in a private way, without the server seing the query in the clear. First we compile the function with Concrete."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "91c766c2-d30a-4e67-bf12-25e4083c67c7",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "Computation Graph\n",
+      "--------------------------------------------------------------------------------\n",
+      "%0 = one_hot_vector        # EncryptedTensor<uint1, shape=(8,)>        ∈ [0, 1]\n",
+      "%1 = database              # ClearTensor<uint5, shape=(8,)>            ∈ [2, 21]\n",
+      "%2 = dot(%0, %1)           # EncryptedScalar<uint5>                    ∈ [2, 21]\n",
+      "return %2\n",
+      "--------------------------------------------------------------------------------\n",
+      "\n",
+      "MLIR\n",
+      "--------------------------------------------------------------------------------\n",
+      "module {\n",
+      "  func.func @main(%arg0: tensor<8x!FHE.eint<5>>, %arg1: tensor<8xi6>) -> !FHE.eint<5> {\n",
+      "    %0 = \"FHELinalg.to_signed\"(%arg0) : (tensor<8x!FHE.eint<5>>) -> tensor<8x!FHE.esint<5>>\n",
+      "    %1 = \"FHELinalg.dot_eint_int\"(%0, %arg1) : (tensor<8x!FHE.esint<5>>, tensor<8xi6>) -> !FHE.esint<5>\n",
+      "    %2 = \"FHE.to_unsigned\"(%1) : (!FHE.esint<5>) -> !FHE.eint<5>\n",
+      "    return %2 : !FHE.eint<5>\n",
+      "  }\n",
+      "}\n",
+      "--------------------------------------------------------------------------------\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "def compile_function(database, show_mlir=False, show_graph=False):\n",
+    "    database_length = database.shape[0]\n",
+    "    inputset_length = 100\n",
```

Comment:
So:
- you're right it could be wrong
- you're right this single one could be sufficient
- but I don't like us advertising about single-element inputset, since people may try to do the same when it's not a good idea

so I added a:

```
# Also add the extreme value, which would actually be sufficient alone, as it reaches the maximal
    # values in get_ith_element_of_database
    inputset.append((make_one_hot_vector(np.argmax(database), database_length), database))
 ```

Good for you?

Reply:
Yes good for me :pray: